index.bs

```diff
@@ -242,6 +244,75 @@ The <dfn attribute for=FileSystemHandle>name</dfn> getter steps are to return [=
   :: Returns true if |handle1| and |handle2| represent the same file or directory.
 </div>
 
+### The {{FileSystemHandle/remove()}} method ### {#api-filesystemhandle-remove}
+
+<div class="note domintro">
+  : await |handle| . {{FileSystemHandle/remove()|remove}}()
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: false })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     Attempting to delete a non-empty directory will result in a promise rejection.
+
+  : await |handle| . {{FileSystemHandle/remove()|remove}}({ {{FileSystemRemoveOptions/recursive}}: true })
+  :: Attempts to remove the entry represented by |handle| from the underlying file system.
+
+     If the entry is a directory, its contents will also be deleted recursively.
+
+     For files or directories with multiple hardlinks or symlinks, the entry which is deleted
+     is the entry corresponding to the path which was used to obtain the handle.
+</div>
+
+<div algorithm>
+The <dfn method for=FileSystemHandle>remove(|options|)</dfn> method steps are:
+
+1. Let |result| be [=a new promise=].
+1. Run these steps [=in parallel=]:
+  1. Let |access| be the result of running [=this=]'s
+     [=FileSystemHandle/entry=]'s [=file system entry/request access=] given
+     "`readwrite`".
+  1. If |access| is not "{{PermissionState/granted}}", reject |result| with a
+     "{{NotAllowedError}}" {{DOMException}} and abort.
+
+  1. Let |entry| be [=this=]'s [=FileSystemHandle/entry=].
+  1. Let |lockResult| be the result of [=file entry/lock/take|taking a lock=]
+     with "`exclusive`" on |entry|.
+  1. If |lockResult| is false, [=reject=] |result| with a
+     "{{NoModificationAllowedError}}" {{DOMException}} and abort.
+
+  1. [=Queue a task=] on the [=storage task source=] to run these steps:
```

Comment:
This should use "queue a storage task"

Reply:
Done. Please let me know if `|entry|'s [=relevant global object=]` is not correct here